### ISSUE TITLE
interesting data structure

### ISSUE THREAD

**Issue by UnixJunkie** [haesbaert/ocaml-mpsc#1]
the opposite would be interesting also:
- single producer
- multiple consumers queue.

**Comment by haesbaert** [comment]
I agree ! I'm just not sure we can just flip it, hopefully we can. Here is the original source:
https://groups.google.com/g/lock-free/c/Vd9xuHrLggE/m/B9-URa3B37MJ?pli=1
